src/main/java/me/desht/pneumaticcraft/common/block/SolarCompressorBlock.java

```diff
@@ -0,0 +1,303 @@
+import net.minecraft.world.phys.shapes.CollisionContext;
+import net.minecraft.world.phys.shapes.Shapes;
+import net.minecraft.world.phys.shapes.VoxelShape;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+import static me.desht.pneumaticcraft.common.util.PneumaticCraftUtils.xlate;
+
+public class SolarCompressorBlock extends AbstractPneumaticCraftBlock implements PneumaticCraftEntityBlock {
+    public static final BooleanProperty BOUNDING = BooleanProperty.create("bounding");
+
+    public SolarCompressorBlock() {
+        super(ModBlocks.defaultProps());
+
+        registerDefaultState(getStateDefinition().any()
+                .setValue(BOUNDING, false));
+    }
+
+    private static final VoxelShape SHAPE = Stream.of(
+            Block.box(0, 1, 4, 16, 16, 12),
+            Block.box(5, 16, 5, 11, 17, 11),
+            Block.box(2, 15, 3, 3, 17, 13),
+            Block.box(13, 15, 3, 14, 17, 13),
+            Block.box(3, 3, 0, 13, 13, 16),
+            Block.box(1, 1, 1, 15, 15, 15),
+            Block.box(0, 0, 0, 16, 1, 16),
+            Block.box(7, 17, 7, 9, 31, 9)
+    ).reduce((v1, v2) -> Shapes.join(v1, v2, BooleanOp.OR)).get();
+
+    @Override
+    public VoxelShape getShape(BlockState state, BlockGetter world, BlockPos pos, CollisionContext context) {
+        if(state.getValue(BOUNDING)) {
+            return Block.box(0,0,0,0,0,0);
+        }
+        else {
+            return SHAPE;
+        }
+    }
+
+    @Override
+    public VoxelShape getCollisionShape(BlockState state, BlockGetter world, BlockPos pos, CollisionContext context) {
+        if(state.getValue(BOUNDING)) {
+            return Block.box(0,0,0,0,0,0);
+        }
+        else {
+            return SHAPE;
+        }
+    }
+
+    @Override
+    protected void createBlockStateDefinition(StateDefinition.Builder<Block, BlockState> builder) {
+        super.createBlockStateDefinition(builder);
+        builder.add(BOUNDING);
+    }
+
+    @Nullable
+    @Override
+    public BlockEntity newBlockEntity(BlockPos pPos, BlockState pState) {
+        return new SolarCompressorBlockEntity(pPos, pState);
+    }
+
+    public BlockPos getMainPos(BlockState state, LevelReader world, BlockPos pos) {
+        // Gets position of main from bounding block based on offset
+        if(state.getValue(BOUNDING)) {
+            SolarCompressorBlockEntity bounding = (SolarCompressorBlockEntity) world.getBlockEntity(pos);
+
+            return pos.subtract(bounding.offsetFromMain);
+        }
+
+        // Returns current position as block is main
+        else {
+            return pos;
+        }
+    }
+
+    @Override
+    public boolean canSurvive(BlockState state, LevelReader world, BlockPos pos) {
+        if (state.getValue(BOUNDING)) {
+            return world.getBlockState(getMainPos(state, world, pos)).getBlock() == this;
+        } else {
+            return world.isEmptyBlock(pos.offset(0,1,0)) &&
```

Comment:
The compressor isn't rotatable around the Y axis? Seems like the panels are always on the +/- Z axis, right?  I think one of the first issues players will raise is to allow it to be rotatable...

Reply:
I don't think it would make sense for the panels to be rotatable since they follow the sun and the sun always has the same path across the sky. The base could be rotatable to allow changing the tube I/O ports' axis though.

Reply:
Yeah, fair point. But the base should be rotatable, definitely.